chainerx_cc/chainerx/cuda/memory_pool.h

```diff
@@ -156,6 +161,11 @@ class MemoryPool {
     cuda_internal::FreeBinsMap free_bins_;  // allocation size => cuda_internal::FreeList
     std::mutex in_use_mutex_;
     std::mutex free_bins_mutex_;
+
+    std::function<void(size_t, MemoryPool&)> malloc_preprocess_hook_;
```

Comment:
I think it's better to place the `MemoryPool&` argument at the beginning, because it's common to all the callbacks.

Reply:
Thank you for comment. Fixed.